inc/saimymac.h

```diff
@@ -0,0 +1,319 @@
+     * @flags CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_PORT, SAI_OBJECT_TYPE_LAG
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_MY_MAC_ATTR_PORT_ID,
+
+    /**
+     * @brief Associated Vlan Id,
+     * if not specified any vlan id will match
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_ONLY
+     * @isvlan true
+     * @default 0
+     */
+    SAI_MY_MAC_ATTR_VLAN_ID,
+
+    /**
+     * @brief MAC Address
+     *
+     * @type sai_mac_t
+     * @flags CREATE_ONLY
+     * @default vendor
+     */
+    SAI_MY_MAC_ATTR_MAC_ADDRESS,
+
+    /**
+     * @brief MAC Address Mask
+     *
+     * @type sai_mac_t
+     * @flags CREATE_ONLY
+     * @default vendor
+     */
+    SAI_MY_MAC_ATTR_MAC_ADDRESS_MASK,
+
+    /**
+     * @brief Enable IPv4 processing
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default true
+     */
+    SAI_MY_MAC_ATTR_V4_ENABLE,
+
+    /**
+     * @brief Enable IPv6 processing
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default true
+     */
+    SAI_MY_MAC_ATTR_V6_ENABLE,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_MY_MAC_ATTR_END,
+
+    /** Custom range base value */
+    SAI_MY_MAC_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_MY_MAC_ATTR_CUSTOM_RANGE_END
+
+} sai_my_mac_attr_t;
+
+/**
+ * @brief Create My MAC entry.
+ *
+ * @param[out] my_mac_id My MAC id
+ * @param[in] switch_id Switch id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_create_my_mac_fn)(
+        _Out_ sai_object_id_t *my_mac_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove My MAC entry
+ *
+ * @param[in] my_mac_id My MAC Id
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_remove_my_mac_fn)(
+        _In_ sai_object_id_t my_mac_id);
+
+/**
+ * @brief Set My MAC entry attribute
+ *
+ * @param[in] my_mac_id My MAC id
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_set_my_mac_attribute_fn)(
+        _In_ sai_object_id_t my_mac_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get My MAC entry attribute
+ *
+ * @param[in] my_mac_id My MAC id
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_my_mac_attribute_fn)(
+        _In_ sai_object_id_t my_mac_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief My MAC table attribute IDs
+ */
+typedef enum _sai_my_mac_table_attr_t
```

Comment:
Can we simplify this table to:

```
/**
 * @brief My MAC entry attribute IDs
 */
typedef enum _sai_my_mac_attr_t
{
    /**
     * @brief Start of attributes
     */
    SAI_MY_MAC_ATTR_START,

    
    /**
     * @brief Object id of the My MAC table
     *
     * @type sai_object_id_t
     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
     * @objects SAI_OBJECT_TYPE_MY_MAC_TABLE
     */
    SAI_MY_MAC_ATTR_TABLE_ID =  SAI_MY_MAC_ATTR_START,

    /**
     * @brief Associated Port, LAG object id,
     * if not specified any port will match
     *
     * @type sai_object_id_t
     * @flags CREATE_ONLY
     * @objects SAI_OBJECT_TYPE_PORT, SAI_OBJECT_TYPE_LAG
     * @allownull true
     * @default SAI_NULL_OBJECT_ID
     */
    SAI_MY_MAC_ATTR_PORT_ID,

    /**
     * @brief Associated Vlan Id,
     * if not specified any vlan id will match
     *
     * @type sai_uint16_t
     * @flags CREATE_ONLY
     * @isvlan true
     * @default 0
     */
    SAI_MY_MAC_ATTR_VLAN_ID,

    /**
     * @brief MAC Address
     *
     * @type sai_mac_t
     * @flags CREATE_ONLY
     * @default vendor
     */
    SAI_MY_MAC_ATTR_MAC_ADDRESS,

    /**
     * @brief MAC Address Mask
     *
     * @type sai_mac_t
     * @flags CREATE_ONLY
     * @default vendor
     */
    SAI_MY_MAC_ATTR_MAC_ADDRESS_MASK,
    
    /**
     * @brief End of attributes
     */
    SAI_MY_MAC_ATTR_END,

    /** Custom range base value */
    SAI_MY_MAC_ATTR_CUSTOM_RANGE_START = 0x10000000,

    /** End of custom range base */
    SAI_MY_MAC_ATTR_CUSTOM_RANGE_END

} sai_my_mac_attr_t;
```

Reply:
I agree with the proposed simplification.

Reply:
Would it be ok to pass in the flags, and let the SAI implementation then program the entry (or not) based on the flags.